apps/engmanagement/src/components/convertkit-subscribe-and-tag.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import {useConvertkit} from '@skillrecordings/convertkit'
+import axios from 'axios'
+import {useFormik} from 'formik'
+import Spinner from 'components/spinner'
+import * as Yup from 'yup'
+
+const ConvertkitSubscribeAndTagForm: React.FC<{tag: string; idNum?: string}> =
+  ({tag, idNum = '1'}) => {
+    const {subscriber} = useConvertkit()
+    const [submitting, setSubmitting] = React.useState<boolean>(false)
+
+    const formik = useFormik({
+      initialValues: {
+        email_address: '',
+        first_name: '',
+      },
+      validationSchema: Yup.object().shape({
+        email_address: Yup.string()
+          .email('Invalid email address')
+          .required('Required'),
+        first_name: Yup.string(),
+      }),
+      enableReinitialize: true,
+      onSubmit: async (values) => {
+        setSubmitting(true)
+        axios
```

Comment:
I'd consider putting this in a serverless function and moving the call to a "service" versus "raw axios" in a callback

note that you are using `async` here and then using more traditional promise API and not `await` - I'd generally try to do one or the other

Reply:
I took this from rustadventure so I'm probably gonna update both of them in next pass if that's cool.
I think we should also consider merging this with default subscribe form from convertkit package.